src/Elastic.Apm/DiagnosticSource/ElasticCoreListeners.cs

```diff
@@ -12,8 +13,8 @@ public class ElasticCoreListeners
 		/// <summary>
 		/// Start listening for diagnosticsource events. Only listens for sources that are part of the Agent.Core package.
 		/// </summary>
-		public void Start() => DiagnosticListener
+		public void Start(AbstractLogger logger) => DiagnosticListener
```

Comment:
Can we default to `Agent.Config.Logger` here and make that parameter optional? That way we would have a less complex public API.

Reply:
I moved these listeners and start behind `IDiagnosticSubscriber` so the public contract folks can just provide us with an instance see e.g:

```diff
- app.UseElasticApm(Configuration);
- new ElasticCoreListeners().Start();
- new ElasticEntityFrameworkCoreListener().Start();
+ app.UseElasticApm(Configuration, new HttpDiagnosticsSubscriber(), new EfCoreDiagnosticsSubscriber());
```

or `IApmAgent` could also have `Subscribe()` to help hook this up even cleaner.

Reply:
Not sure we are ok here. The downside I think we still have is that once this is not started in the `ApmMiddlewareExtension` then activating those is still more complex then it should be - I'm not sure it's even possible now. Example scenario: .NET Core console application that wants to monitor outgoing `HttpClient` calls. In that case it would be nice to have the option to subscribe to those diagnostic source events (aka turn on outgoing HTTP monitoring) with a 1 liner. And that was `new ElasticCoreListeners().Start();` before this change. I think we should aim for something that is as simple as this one because it's a very basic feature and we can't be sure that we can turn this on automatically in all scenarios. Same applies to other thing (EFCore, etc..)